libs/application-generic/src/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -49,30 +60,138 @@ export class GetSubscriberPreference {
       '',
       {
         _organization: command.organizationId,
-        templatesSize: templateList.length,
+        templatesSize: workflowList.length,
       },
     );
 
-    // TODO: replace this runtime mapping with a single query to the database
-    const subscriberWorkflowPreferences = await Promise.all(
-      templateList.map(async (template) =>
-        this.getSubscriberTemplatePreferenceUsecase.execute(
-          GetSubscriberTemplatePreferenceCommand.create({
-            organizationId: command.organizationId,
-            subscriberId: command.subscriberId,
-            environmentId: command.environmentId,
-            template,
-            subscriber,
-            includeInactiveChannels: command.includeInactiveChannels,
-          }),
-        ),
-      ),
+    const allPreferences = await this.preferencesRepository.find({
+      _environmentId: command.environmentId,
+      $or: [
+        {
+          _subscriberId: subscriber._id,
+          type: PreferencesTypeEnum.SUBSCRIBER_WORKFLOW,
+        },
+        {
+          _templateId: { $in: workflowList.map((wf) => wf._id) },
+          type: {
+            $in: [
+              PreferencesTypeEnum.WORKFLOW_RESOURCE,
+              PreferencesTypeEnum.USER_WORKFLOW,
+            ],
+          },
+        },
+        {
+          _subscriberId: subscriber._id,
+          type: PreferencesTypeEnum.SUBSCRIBER_GLOBAL,
+        },
+      ],
+    });
+
+    const subscriberGlobalPreferences = allPreferences.filter(
+      (preference) =>
+        preference._subscriberId === subscriber._id &&
+        preference.type === PreferencesTypeEnum.SUBSCRIBER_GLOBAL,
+    );
+
+    const workflowPreferenceSets = allPreferences.reduce<
```

Comment:
I have split into a single query for each preference, the code reads nicer and the size of the collection means it will be more performant to query in parallel.